src/module.cpp

```diff
@@ -3186,6 +3187,8 @@ int Module::CompileAndOutput(const char *srcFile, Arch arch, const char *cpu, st
         delete g->target;
         g->target = nullptr;
 
+        delete targetMachine;
```

Comment:
A pointer to `llvm::TargetMachine` is owned by `Target`, why not deleting it in its destructor?

Reply:
Unfortunately, `Target` doesn't really own `llvm::TargetMachine *m_targetMachine`, see comment here: https://github.com/ispc/ispc/blob/f080d39c99decb9947e7fa5688a56773f841fe11/src/ispc.h#L354

and example of usage after `Target` object deconstruction:
- `Target` created https://github.com/ispc/ispc/blob/f080d39c99decb9947e7fa5688a56773f841fe11/src/module.cpp#L3247
- Ownership of `m_targetMachine` is passed to the `targetMachines` https://github.com/ispc/ispc/blob/f080d39c99decb9947e7fa5688a56773f841fe11/src/module.cpp#L3260
- `Target` destroyed https://github.com/ispc/ispc/blob/f080d39c99decb9947e7fa5688a56773f841fe11/src/module.cpp#L3321
- `targetMachine` is used after that here https://github.com/ispc/ispc/blob/f080d39c99decb9947e7fa5688a56773f841fe11/src/module.cpp#L3340

Reply:
Ok, that's caused by messy way of doing multi-target compilation.

Reply:
I've had a deeper look and I think this needs to be cleaned up. Specifically:
- `TargetMachine` really belongs to the `Target` and needs to be destroyed in its destructor.
- The cases when it escapes `Target` lifetime are not legitimate:
  - `targetMachines[]` array should be a bool array with a name "CompiledTargets" or something like that.
  - the use of `firstTargetMachine` should be turned into `g->target->GetTargetMachine()`, as for the dispatch module we create a new `Target` anyway and it creates a `TargetMachine` as well - and this one needs to be used.

I'll proceed with merging this PR, and will create an issue to follow up with the clean up explained here.

Reply:
I've created https://github.com/ispc/ispc/issues/2578 to follow up.